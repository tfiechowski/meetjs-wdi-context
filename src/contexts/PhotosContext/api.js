import photos from "./apiData";
console.log("photos ", photos);

export default {
  fetchPhotos: () => {
    return photos;
  }
};
