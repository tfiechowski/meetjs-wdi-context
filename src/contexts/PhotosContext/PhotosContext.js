import React, { Component, createContext } from "react";
import api from "./api";

const Context = createContext({});

export class PhotosProvider extends Component {
  constructor(props) {
    super(props);

    this.state = {
      photos: [],
      fetchPhotos: this.fetchPhotos
    };
  }

  render() {
    return (
      <Context.Provider value={this.state}>
        {this.props.children}
      </Context.Provider>
    );
  }

  fetchPhotos = async () => {
    const photos = await api.fetchPhotos();
    this.setState(state => ({ photos: [state.photos, ...photos] }));
  };
}

export default Context;
