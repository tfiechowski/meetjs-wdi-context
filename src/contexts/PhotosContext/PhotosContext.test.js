import { mount } from "enzyme";
import React from "react";
import { PhotosProvider } from "./PhotosContext";

describe("PhotosContext", () => {
  it("Should have empty photos list by default", () => {
    const wrapper = mount(<PhotosProvider />);
    expect(wrapper.state().photos).toEqual([]);
  });

  it("Should properly update photos after fetching", async () => {
    const wrapper = mount(<PhotosProvider />);
    await wrapper.instance().fetchPhotos();
    const { photos } = wrapper.state();

    expect(photos).toMatchSnapshot();
  });
});
