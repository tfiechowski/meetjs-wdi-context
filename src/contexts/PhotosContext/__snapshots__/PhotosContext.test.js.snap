// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotosContext Should properly update photos after fetching 1`] = `
Array [
  Array [],
  Object {
    "albumId": 1,
    "id": 1,
    "thumbnailUrl": "https://via.placeholder.com/150/92c952",
    "title": "accusamus beatae ad facilis cum similique qui sunt",
    "url": "https://via.placeholder.com/600/92c952",
  },
  Object {
    "albumId": 1,
    "id": 2,
    "thumbnailUrl": "https://via.placeholder.com/150/771796",
    "title": "reprehenderit est deserunt velit ipsam",
    "url": "https://via.placeholder.com/600/771796",
  },
  Object {
    "albumId": 1,
    "id": 3,
    "thumbnailUrl": "https://via.placeholder.com/150/24f355",
    "title": "officia porro iure quia iusto qui ipsa ut modi",
    "url": "https://via.placeholder.com/600/24f355",
  },
  Object {
    "albumId": 1,
    "id": 4,
    "thumbnailUrl": "https://via.placeholder.com/150/d32776",
    "title": "culpa odio esse rerum omnis laboriosam voluptate repudiandae",
    "url": "https://via.placeholder.com/600/d32776",
  },
  Object {
    "albumId": 1,
    "id": 5,
    "thumbnailUrl": "https://via.placeholder.com/150/f66b97",
    "title": "natus nisi omnis corporis facere molestiae rerum in",
    "url": "https://via.placeholder.com/600/f66b97",
  },
  Object {
    "albumId": 1,
    "id": 6,
    "thumbnailUrl": "https://via.placeholder.com/150/56a8c2",
    "title": "accusamus ea aliquid et amet sequi nemo",
    "url": "https://via.placeholder.com/600/56a8c2",
  },
  Object {
    "albumId": 1,
    "id": 7,
    "thumbnailUrl": "https://via.placeholder.com/150/b0f7cc",
    "title": "officia delectus consequatur vero aut veniam explicabo molestias",
    "url": "https://via.placeholder.com/600/b0f7cc",
  },
]
`;
